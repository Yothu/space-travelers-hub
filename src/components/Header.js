import { NavLink } from 'react-router-dom';
import styled from 'styled-components';
import pageLogo from '../assets/images/planet.png';

const Container = styled.header`
  padding: 1rem;
  border-bottom: 2px solid black;
`;

const PageInfo = styled.div`
  display: flex;
`;

const LogoStyle = {
  height: '4.8rem',
  width: '4.8rem',
};

const LinksContainer = styled.div`
  display: flex;
  align-items: center;
`;

const InnerContainer = styled.nav`
  display: flex;
  justify-content: space-between;
`;

const Header = () => {
  const PageLogo = () => <img src={pageLogo} style={LogoStyle} alt="page-logo" />;
  const title = 'Space Travelers\' Hub';

  return (
    <Container>
      <InnerContainer>
        <PageInfo>
          <PageLogo />
          <h1>{title}</h1>
        </PageInfo>
        <LinksContainer>
          <NavLink
            to="/"
            style={({ isActive }) => ({
              color: isActive ? '#000' : '#545e6f',
              textDecoration: isActive ? 'underline' : 'none',
            })}
          >
            Rockets
          </NavLink>
          <NavLink
            to="/profile"
            style={({ isActive }) => ({
              color: isActive ? '#000' : '#545e6f',
              textDecoration: isActive ? 'underline' : 'none',
            })}
          >
            My Profile
          </NavLink>
        </LinksContainer>
      </InnerContainer>
    </Container>
  );
};

export default Header;
