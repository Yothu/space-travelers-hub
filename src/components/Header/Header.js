import { NavLink } from 'react-router-dom';
import styled from 'styled-components';
import pageLogo from '../../assets/images/planet.png';
import '../Font.css';

const Container = styled.header`
  padding: 0.5rem 1rem;
  border-bottom: 1px solid lightgray;
`;

const PageInfo = styled.div`
  display: flex;
  align-items: center;
`;

const LogoStyle = {
  width: '3rem',
  height: '3rem',
  paddingLeft: '1rem',
  paddingRight: '1rem',
};

const Title = styled.h1`
  margin: 0;
  font-weight: 400;
  font-size: 1.5rem;
  font-family: Arial, Helvetica, sans-serif !important;
`;

const LinksContainer = styled.div`
  display: flex;
  align-items: center;
  padding-right: 1rem;
`;

const InnerContainer = styled.nav`
  display: flex;
  justify-content: space-between;
`;

const Header = () => {
  const PageLogo = () => <img src={pageLogo} style={LogoStyle} alt="page-logo" />;
  const title = 'Space Travelers\' Hub';

  return (
    <Container>
      <InnerContainer>
        <PageInfo>
          <PageLogo />
          <Title className="font-1">{title}</Title>
        </PageInfo>
        <LinksContainer>
          <NavLink
            to="/"
            style={({ isActive }) => ({
              color: '#000FFF',
              padding: '0 0.5rem',
              fontSize: '0.8rem',
              textDecoration: isActive ? 'underline' : 'none',
            })}
          >
            Rockets
          </NavLink>
          <NavLink
            to="/missions"
            style={({ isActive }) => ({
              color: '#000FFF',
              padding: '0 0.5rem',
              fontSize: '0.8rem',
              textDecoration: isActive ? 'underline' : 'none',
            })}
          >
            Missions
          </NavLink>
          <NavLink
            to="/profile"
            style={({ isActive }) => ({
              color: '#000FFF',
              padding: '0 0 0 0.5rem',
              fontSize: '0.8rem',
              borderLeft: '1px solid #000',
              textDecoration: isActive ? 'underline' : 'none',
            })}
          >
            My Profile
          </NavLink>
        </LinksContainer>
      </InnerContainer>
    </Container>
  );
};

export default Header;
