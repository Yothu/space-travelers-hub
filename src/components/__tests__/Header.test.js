import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import React from 'react';
import Header from '../Header/Header';

describe('Rockets page tests', () => {
  it('it renders correctly', () => {
    const header = renderer
      .create(
        <BrowserRouter>
          <Header />
        </BrowserRouter>,
      )
      .toJSON();
    expect(header).toMatchSnapshot();
  });
});
