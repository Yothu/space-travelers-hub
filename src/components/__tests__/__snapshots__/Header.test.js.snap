// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets page tests it renders correctly 1`] = `
<header
  className="sc-bdvvtL cQxWel"
>
  <nav
    className="sc-eCImPb bmNDXF"
  >
    <div
      className="sc-gsDKAQ hznLYe"
    >
      <img
        alt="page-logo"
        src="planet.png"
        style={
          Object {
            "height": "3rem",
            "paddingLeft": "1rem",
            "paddingRight": "1rem",
            "width": "3rem",
          }
        }
      />
      <h1
        className="sc-dkPtRN EgFWJ font-1"
      >
        Space Travelers' Hub
      </h1>
    </div>
    <div
      className="sc-hKwDye gUBtGx"
    >
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={
          Object {
            "color": "#000FFF",
            "fontSize": "0.8rem",
            "padding": "0 0.5rem",
            "textDecoration": "underline",
          }
        }
      >
        Rockets
      </a>
      <a
        className=""
        href="/missions"
        onClick={[Function]}
        style={
          Object {
            "color": "#000FFF",
            "fontSize": "0.8rem",
            "padding": "0 0.5rem",
            "textDecoration": "none",
          }
        }
      >
        Missions
      </a>
      <a
        className=""
        href="/profile"
        onClick={[Function]}
        style={
          Object {
            "borderLeft": "1px solid #000",
            "color": "#000FFF",
            "fontSize": "0.8rem",
            "padding": "0 0 0 0.5rem",
            "textDecoration": "none",
          }
        }
      >
        My Profile
      </a>
    </div>
  </nav>
</header>
`;
